nearquake/data_processor.py

```
@@ -239,59 +243,47 @@ def _fetch_location_detail(self, event) -> LocationDetails:
-        if upload_type not in {"single", "batch"}:
-            _logger.error(
-                f"Invalid upload type '{upload_type}'. Supported types are 'single' or 'batch'."
-            )
-            raise ValueError(
-                f"Invalid upload type '{upload_type}'. Supported types are 'single' or 'batch'."
-            )
+        try:
+            datetime.strptime(start_date, "%Y-%m-%d").date()
+            datetime.strptime(end_date, "%Y-%m-%d").date()
+        except ValueError:
+            raise ValueError("Invalid date format. Use YYYY-MM-DD.")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)